src/main/java/seedu/duke/chatbot/Storage.java

```diff
@@ -0,0 +1,176 @@
+package seedu.duke.chatbot;
+
+import seedu.duke.exceptions.DukeException;
+import seedu.duke.exceptions.UnableToUpdateDatabaseException;
+import seedu.duke.task.Deadline;
+import seedu.duke.task.Event;
+import seedu.duke.task.Task;
+import seedu.duke.task.TaskList;
+import seedu.duke.task.ToDo;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.TemporalAccessor;
+import java.util.Scanner;
+
+/**
+ * Handles updating of database, a txt file, whenever {@link TaskList} is updated.
+ */
+public class Storage {
+    private final String filePath;
+    private final Ui ui;
+
+    /**
+     * Creates a Storage object which is used to update the database as commands are executed.
+     * @param filePath specifying the path to the database
+     * @param ui which is used to handle printing of replies to user
+     */
+    Storage(String filePath, Ui ui) {
+        this.filePath = filePath;
+        this.ui = ui;
+    }
+
+    public String getFilePath() {
+        return this.filePath;
+    }
+
+    /**
+     * Creates a {@link TaskList} based on what was saved in the database from previous run of Duke.
+     * @return {@link TaskList} with all previous tasks added
+     * @throws DukeException if database file is not found or unable to create the {@link TaskList}
+     */
+    TaskList getOldTaskList() throws DukeException {
```

Comment:
Consider breaking this method up into multiple sub methods to help with nesting and readability! Also, access permission tags like "protected" or "private" could be used if you choose to break it up!

Reply:
I edited and broke it down using one more sub-method , thanks for catching this